### ISSUE TITLE
Level guards

### ISSUE THREAD

**Issue by Asarew** [rs/zerolog#31]
One thing that i believe is missing is the possibly to get the current level of a `zerolog.Logger`. This might be useful when constructing log messages might be expensive (hot path + allocating) in certain places.
Something like
```golang
func (l *Logger) Is(lvl Level) bool {
        return l.level >= lvl
}
```
Don't know if this is something interesting because the only logging package that i could (quickly) find that has something like this is https://github.com/mgutz/logxi.

**Comment by rs** [comment]
Do you have a use-case in mind?

**Comment by Asarew** [comment]
I'm currently developing a system where i heavily rely on logging to debug e2e tests. The places where i have to log items are hot-paths in my execution. When constructing the log Event I first log an `interface{}` and grab `time.Now()`, the operation it self is asynchronous and returns a wait object (future/promise/deferred).
When i wan to log the duration of the operation; i have to start a goroutine and wait on object to return.
But when to the logger level is higher than debug it's unnecessary to create a goroutine everytime where the result of the func is to discard the message anyway.

This was my specific use-case but i see the same pattern in other projects as well like kubernetes with there use of the glog package: https://godoc.org/github.com/golang/glog#V

**Comment by rs** [comment]
With `zerolog` you can just do this:

```go
if e := log.Debug(); e.Enabled() {
    // Compute log output only if current level is debug or higher.
    value := compute()
    e.Str("foo": value).Msg("some debug message")
}
```

**Comment by Asarew** [comment]
Thnx for the quick solution/reponse! I hadn't spotted the `Event.Enabled()` method. Even though i'm not the biggest fan of this api, it does exactly what i was looking for! closing this issue as answered :+1:
